lib/modules/manager/maven-wrapper/artifacts.ts

```
@@ -0,0 +1,205 @@
+    ).filter(is.truthy);
+
+    logger.debug(
+      { files: updateArtifactsResult.map((r) => r.file?.path) },
+      `Returning updated maven-wrapper files`
+    );
+    return updateArtifactsResult;
+  } catch (err) {
+    logger.debug({ err }, 'Error setting new Maven Wrapper release value');
+    return [
+      {
+        artifactError: {
+          lockFile: packageFileName,
+          stderr: err.message,
+        },
+      },
+    ];
+  }
+}
+
+async function getUpdatedArtifacts(
+  status: StatusResult,
+  artifactFileNames: string[]
+): Promise<(UpdateArtifactsResult | null)[]> {
+  const updatedResults: (UpdateArtifactsResult | null)[] = [];
+  for (const artifactFileName of artifactFileNames) {
+    updatedResults.push(await addIfUpdated(status, artifactFileName));
+  }
+  return updatedResults;
+}
+
+/**
+ * Find compatible java version for maven.
+ * see https://maven.apache.org/developers/compatibility-plan.html
+ * @param mavenWrapperVersion current maven version
+ * @returns A Java semver range
+ */
+export function getJavaConstraint(
+  mavenWrapperVersion: string | null | undefined
+): string | null {
+  const major = mavenWrapperVersion
+    ? mavenVersioning.getMajor(mavenWrapperVersion)
+    : null;
+
+  if (major && major >= 3) {
+    return '^17.0.0';
+  }
+
+  return '^8.0.0';
+}
+
+async function executeWrapperCommand(
+  cmd: string,
+  config: UpdateArtifactsConfig
+): Promise<void> {
+  logger.debug(`Updating maven wrapper: "${cmd}"`);
+  const execOptions: ExecOptions = {
+    docker: {
+      image: 'java',
+      tagConstraint:
+        config.constraints?.java ?? getJavaConstraint(config.currentValue),
+      tagScheme: semver,
+    },
+  };
+
+  try {
+    await exec(cmd, execOptions);
+  } catch (err) {
+    // istanbul ignore if
+    if (err.message === TEMPORARY_ERROR) {
+      throw err;
+    }
+    logger.warn(
+      { err },
+      'Error executing maven wrapper update command. It can be not a critical one though.'
```

shouldn't we return an artifact error, so the user is aware of a failure?